atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderRole.java

```diff
@@ -228,7 +228,13 @@ private void stepDown() {
   /** Sets the current node as the cluster leader. */
   private void takeLeadership() {
     raft.setLeader(raft.getCluster().getLocalMember().memberId());
-    raft.getCluster().getRemoteMemberStates().forEach(m -> m.openReplicationContext(raft.getLog()));
+    raft.getCluster()
+        .getRemoteMemberStates()
+        .forEach(
+            member -> {
+              member.openReplicationContext(raft.getLog());
+              appender.observeRemainingMemberEntries(member);
```

Comment:
:wrench: Not sure if this is really necessary here. On becoming leader, follower's state is reset so the matchIndex will be 0.  So the gap repeated by the metrics will be always high immediately after leader change, even though the followers are almost up-to-date.

Reply:
I added it for the reset on cluster restart or something like that. But you're right that it will always `lastIndex`, so it's not that much more useful - I guess what I wanted is to reset it to 0. Should we just do that?

Reply:
Setting to 0 would be fine as well. Or we don't have to set anything at all. Then grafana shows no value until the first AppendRequest is acknowledged, and that is ok I guess.

Reply:
Let's remove it